frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -111,37 +111,57 @@ export const eventListener: TelemetryEventListener = async (
   properties?: any,
 ) => {
   if (TELEMETRY_DISABLED || !apiKey) {
-    // eslint-disable-next-line no-console
-    console.debug(
-      'console-telemetry-plugin: telemetry disabled - ignoring telemetry event:',
-      eventType,
-      properties,
-    );
-    return;
-  }
-  if (TELEMETRY_DEBUG) {
-    // eslint-disable-next-line no-console
-    console.debug('console-telemetry-plugin: received telemetry event:', eventType, properties);
+    if (TELEMETRY_DEBUG) {
+      // eslint-disable-next-line no-console
+      console.debug(
+        'console-telemetry-plugin: telemetry disabled - ignoring telemetry event:',
+        eventType,
+        properties,
+      );
+    }
     return;
   }
   switch (eventType) {
     case 'identify':
       {
         const { user, ...otherProperties } = properties;
-        const id = user?.metadata?.name;
-        if (id) {
+        const clusterId = otherProperties?.clusterId;
+        const organizationId = otherProperties?.organizationId;
+        const username = user?.metadata?.name;
+        if (username) {
+          let anonymousIdInput: string;
+          if (organizationId) {
+            if (username === 'kubeadmin' || username === 'kube:admin') {
+              anonymousIdInput = `${organizationId}@${clusterId}`;
```

Comment:
is there a format the ID should meet ?
can we just do `<user>:<clusterID>:<orgID>` ? that way the format is unified and if the orgID will be missing then we know the anonymousID ends with `:`

Reply:
The string is hashed after that. The @ sign here isn't needed. I can also just concat both strings. It will generate an "anonymous" id in a way that we can identify the same user on the next day, without identify the user itself.

The goal is to
1. identify kubeadmins within an org on different clusters. That's the reason for this if condition here.
2. For all other users, we generate a unique hash for each organization, so that we will have the same hash for a user on different clusters.

We discussed this last week with @spadgett, Ali and Parag, see https://docs.google.com/document/d/11gxr_7kxMqm1zSJC5pPVHZhvLJthZGLH-QvqCqAyFdc/edit#heading=h.g6v969t9p2hf